src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
+        str += `${this[i]}`;
+      }
```

Comment:
(Just recommendation) I would move it into variable
```suggestion
    for (let i = 0; i < this.length; i++) {
      const element = this[i];
      
      if (element !== undefined && element !== null) {
        str += `${element}`;
      }
```